multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/OperationTimeAggregator.java

```diff
@@ -9,52 +10,53 @@
-                                                                                                             processTimeCalculator::calculate));
-        processTimesForSubProcesses.forEach((key, value) -> logProcessTimeIndividually(value, correlationId,
-                key));
-
-        ProcessTime rootProcessTime = processTimesForSubProcesses.get(correlationId);
-        logOverallProcesstime(correlationId, rootProcessTime, processTimesForSubProcesses.values());
+        return historicSubProcesses.stream()
+                                   .collect(Collectors.toMap(processId -> processId, processTimeCalculator::calculate));
     }
 
-    private void logProcessTimeIndividually(ProcessTime processTime, String correlationId, String processInstanceId) {
-        ProcessTimeLogger.logProcessTimeIndividually(LOGGER, processTime, correlationId, processInstanceId);
-    }
+    public ProcessTime computeOverallProcessTime(String correlationId, Map<String, ProcessTime> processTimes) {
+        //may not be the best idea if this changes the input parameter
+        ProcessTime rootProcessTime = processTimes.remove(correlationId);
 
-    private void logOverallProcesstime(String correlationId, ProcessTime rootProcessTime,
-                                       Collection<ProcessTime> subProcessesProcessTimes) {
-        long overallDelayBetweenSteps = subProcessesProcessTimes.stream()
-                                                                .mapToLong(ProcessTime::getDelayBetweenSteps)
-                                                                .sum();
-        ProcessTime overallProcessTime = ImmutableProcessTime.copyOf(rootProcessTime)
-                                                             .withDelayBetweenSteps(rootProcessTime.getDelayBetweenSteps()
-                                                                 + overallDelayBetweenSteps);
+        long overallDelayBetweenSteps = processTimes.values()
+                                                    .stream()
+                                                    .mapToLong(ProcessTime::getDelayBetweenSteps)
+                                                    .sum();
+
+        return ImmutableProcessTime.copyOf(rootProcessTime)
+                                   .withDelayBetweenSteps(rootProcessTime.getDelayBetweenSteps() + overallDelayBetweenSteps);
+    }
 
-        ProcessTimeLogger.logOverallProcessTime(LOGGER, overallProcessTime, correlationId);
+    public boolean hasParallelSubProcesses(Map<String, ProcessTime> subProcessTimes) {
+        ProcessTime[] processTimes = subProcessTimes.values()
+                                                    .toArray(new ProcessTime[0]);
+        Arrays.sort(processTimes, Comparator.comparing(ProcessTime::getStartTime));
+
+        for (int i = 0; i < processTimes.length - 1; i++) {
+            for (int j = i + 1; j < processTimes.length; j++) {
+                if (processTimes[i].getEndTime()
```

Comment:
That won't be problem because the process times operated on here are only the sub processes of the process on which we are calculating the delay. The parent process is absent in this array.